core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -1131,6 +1337,230 @@ TEST(TEST_CATEGORY, mathematical_functions_absolute_value) {
+struct TestFloatingPointRemainderFunction : FloatingPointComparison {
+  TestFloatingPointRemainderFunction() { run(); }
+  void run() const {
+    int errors = 0;
+    Kokkos::parallel_reduce(Kokkos::RangePolicy<Space>(0, 1), *this, errors);
+    ASSERT_EQ(errors, 0);
+  }
+  KOKKOS_FUNCTION void operator()(int, int& e) const {
+    using Kokkos::fmod;
+    if (!compare(fmod(6.2f, 4.f), 2.2f, 1) &&
+        !compare(fmod(-6.2f, 4.f), -2.2f, 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed fmod(float)\n");
+    }
+    if (!compare(
+            fmod(static_cast<KE::half_t>(6.2f), static_cast<KE::half_t>(4.f)),
+            static_cast<KE::half_t>(2.2f), 1) &&
+        !compare(
+            fmod(static_cast<KE::half_t>(-6.2f), static_cast<KE::half_t>(4.f)),
+            -static_cast<KE::half_t>(2.2f), 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed fmod(KE::half_t)\n");
+    }
+    if (!compare(
+            fmod(static_cast<KE::bhalf_t>(6.2f), static_cast<KE::bhalf_t>(4.f)),
+            static_cast<KE::bhalf_t>(2.2f), 1) &&
+        !compare(fmod(static_cast<KE::bhalf_t>(-6.2f),
+                      static_cast<KE::bhalf_t>(4.f)),
+                 -static_cast<KE::bhalf_t>(2.2f), 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed fmod(KE::bhalf_t)\n");
+    }
+    if (!compare(fmod(6.2, 4.), 2.2, 1) && !compare(fmod(-6.2, 4.), -2.2, 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed fmod(double)\n");
+    }
+#ifdef MATHEMATICAL_FUNCTIONS_HAVE_LONG_DOUBLE_OVERLOADS
+    if (!compare(fmod(6.2l, 4.l), 2.2l, 1) &&
+        !compare(fmod(-6.2l, 4.l), -2.2l, 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed fmod(long double)\n");
+    }
+#endif
+    // special values
+    using Kokkos::isinf;
+    using Kokkos::isnan;
+    if (!isinf(fmod(-KE::infinity<float>::value, 1.f)) &&
+        !isnan(fmod(-KE::quiet_NaN<float>::value, 1.f))) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF(
+          "failed fmod(floating_point) special values\n");
+    }
+
+    static_assert(std::is_same<decltype(fmod(static_cast<KE::half_t>(4.f),
+                                             static_cast<KE::half_t>(4.f))),
+                               KE::half_t>::value,
+                  "");
+    static_assert(std::is_same<decltype(fmod(static_cast<KE::bhalf_t>(4.f),
+                                             static_cast<KE::bhalf_t>(4.f))),
+                               KE::bhalf_t>::value,
+                  "");
+    static_assert(std::is_same<decltype(fmod(4.f, 4.f)), float>::value, "");
+    static_assert(std::is_same<decltype(fmod(5., 5.)), double>::value, "");
+#ifdef MATHEMATICAL_FUNCTIONS_HAVE_LONG_DOUBLE_OVERLOADS
+    static_assert(std::is_same<decltype(fmod(6.l, 6.l)), long double>::value,
+                  "");
+#endif
+  }
+};
+
+TEST(TEST_CATEGORY, mathematical_functions_remainder_function) {
+  TestFloatingPointRemainderFunction<TEST_EXECSPACE>();
+}
+
+#if 0
```

Comment:
???

Reply:
I could not get this to work for either built-in types of half types. Would you like me to file a issue to expand our math function test coverage?

Reply:
Please open an issue. 

I would also prefer a variadic function instead of the variadic macro but otherwise, the pull request is fine with me (we still need to document it well though).

Reply:
https://github.com/kokkos/kokkos/issues/6275